lib/nrf_modem_lib/Kconfig.modemlib

```diff
@@ -10,28 +10,6 @@ config HEAP_MEM_POOL_SIZE
 	int
 	default 512
 
-config NRF_MODEM_LIB_SYS_INIT
-	bool "Initialize during SYS_INIT [DEPRECATED]"
-	select DEPRECATED
-	help
-	  Initialize the Modem library automatically during the SYS_INIT sequence.
-	  Please note that initialization is synchronous and can take up to one
-	  minute in case the modem firmware is updated.
-	  This Kconfig option is deprecated. The application should instead use the
-	  nrf_modem_lib_init() function to initialize the modem library.
-
-config NRF_MODEM_LIB_NET_IF_AUTO_START
-	bool "Network interface auto-start"
-	depends on NRF_MODEM_LIB_NET_IF
-	default y
-	help
-	  If this option is enabled, the modem is automatically initialized from the main context
-	  at SYS init.
-	  Initialization of the modem might take some time, so bear in mind that the main context
-	  will be blocked for that duration.
-	  If this option is disabled the modem must be manually initialized by bringing
-	  the network interface's administrative state up by calling net_if_up().
```

Comment:
Sorry I missed this in my earlier reviews, I was confused by the naming and description -- Do not remove this, it has nothing to do with `NRF_MODEM_LIB_SYS_INIT`

It is a convenience KConfig that controls the value of `NET_IF_NO_AUTO_START` for the LTE iface. We definitely don't want to get rid of it. 

The description in KConfig is a bit misleading -- It's technically correct because `net_if_up` does initialize the modem, and  `NET_IF_NO_AUTO_START` causes `net_if_up` to be called by Zephyr on boot, but the setting is really about controlling `NET_IF_NO_AUTO_START`.

Reply:
See this line for where it is used:
https://github.com/nrfconnect/sdk-nrf/blob/b88446c2298cc8bc4dd4fabcdb12a03d5cf8de32/lib/nrf_modem_lib/nrf91_sockets.c#L1107

Reply:
> It's technically correct because net_if_up does initialize the modem, and NET_IF_NO_AUTO_START causes net_if_up to be called by Zephyr on boot

There is a big problem with that. I have tried to explain it in the comment below.